src/agent/ICheqd.ts

```diff
@@ -1919,13 +2210,176 @@ export class Cheqd implements IAgentPlugin {
                 resourceMetadata: topArgs?.returnStatusListMetadata ? await Cheqd.fetchStatusList2021Metadata(credential) : undefined
             } satisfies RevocationResult
         } catch (error) {
-            // silent fail + early exit, optimised for parallelisation, use with Promise.allSettled
+            // silent fail + early exit
             console.error(error)
 
             return { revoked: false, error: error as IError } satisfies RevocationResult
         }
     }
 
+    static async revokeCredentials(credentials: VerifiableCredential[], options?: ICheqdStatusList2021Options): Promise<BulkRevocationResult> {
```

Comment:
I think using revocationOptions here is better than credentials.

instead of converting revocationOptions -> credential, we can convert credential -> revocationOptions.

Bulk revocation just needs to revoke multiple indices in statuslist. Accepting separate credentials where they can have a same index or can be part of different statuslist doesn't look like a good approach to me

Reply:
Top-level methods handle validation.

Nested / underlying functions are utilities that don't know or validate internal state, hence why they are static.

The whole library has been designed with that in mind.


Let me know if you have questions.